### PR TITLE
GLFW -> SDL2

### PR DESCRIPTION
**Pull request by BeauSLM**
Migrate from GLFW to SDL2. This has a number of benefits:

- SDL2 provides everything glfw does, along with several additional useful features (see https://wiki.libsdl.org/SDL2/Introduction)
    - threading api, actually good timer, etc
- the api is considerably more straightforward and simple to use
    - e.g. no callbacks
- got to clean out a lot of code including callback code. Windowing and input handling look so much better!
- easy vsync
- etc.

Insert obligatory "test on your machines so I can fix and then merge" here. Leave a review to lmk if everything worked or if there was as problem.

### CODE REVIEW

**Comment by BeauSLM** [comment]
Good news: runs on my windows machine

Bad news: idk why vsync isn't working, according to my MSI afterburner framerate counter

**Review by SamuelGarm — APPROVED** [review]
Yup, it works

File: code/client/Window.cpp [hunk 1]
```diff
@@ -4,66 +4,37 @@
...
+	SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_CORE);
+	SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
+	SDL_GL_SetAttribute(SDL_GL_CONTEXT_FLAGS, SDL_GL_CONTEXT_DEBUG_FLAG);
+	SDL_GL_SetAttribute(SDL_GL_CONTEXT_FLAGS, SDL_GL_CONTEXT_FORWARD_COMPATIBLE_FLAG);
+
+	window = std::unique_ptr<SDL_Window, WindowDeleter>(SDL_CreateWindow("CPSC 453", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, width, height, SDL_WINDOW_OPENGL));
+	if (!window)
+		sdl_error_handler(1);
+
+	// TODO:(beau) clean up
```
Comment (dylanleclair) [inline code/client/Window.cpp:31]: does this need to be cleaned up still?

File: code/client/Window.cpp [hunk 2]
```diff
@@ -72,47 +43,28 @@ Window::Window(
...
-	if (callbacks != nullptr) {
-		connectCallbacks();
-	}
-}
-
-
-Window::Window(int width, int height, const char* title, GLFWmonitor* monitor, GLFWwindow* share)
-	: Window(nullptr, width, height, title, monitor, share)
-{}
+	// TODO(beau): put this somewhere obvious
```
Comment (dylanleclair) [inline code/client/Window.cpp:46]: todo?

**Review by dylanleclair — COMMENTED** [review]
Are we able to remove GLFW as a dependency if it's no longer used?

File: code/client/Window.cpp (hunk 1 above)
Comment (BeauSLM) [reply code/client/Window.cpp:31]: I spoke too soon. There's some other things we could be doing to clean up sdl as well, but atm having the os clean up when the process ends should work fine. It's another situation where I think where stuff should go will become clear later. I'll put a long note in the window class that enumerates the sdl cleanup things we should do.

e.g. run SDL_Qui() with atexit? with the window destructor? what about cleaning up or opengl context? will we change contexts? clean up in the destructor? also things have to run in a certain order or weird shit happens so we'll probably have to look into that, etc.

**Comment by BeauSLM** [comment]
> Are we able to remove GLFW as a dependency if it's no longer used?

Yep just now lmao

**Review by dylanleclair — APPROVED** [review]
looks good to me

**Pull request merged**